### PR TITLE
Merge Symfony 7 into 2.0

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ChangeShopUserPasswordHandler.php [hunk 1]
```diff
@@ -17,10 +17,11 @@
 use Sylius\Component\Core\Model\ShopUserInterface;
 use Sylius\Component\User\Repository\UserRepositoryInterface;
 use Sylius\Component\User\Security\PasswordUpdaterInterface;
-use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
 use Webmozart\Assert\Assert;
 
-final readonly class ChangeShopUserPasswordHandler implements MessageHandlerInterface
+#[AsMessageHandler]
+final readonly class ChangeShopUserPasswordHandler
```
Comment (jakubtobiasz) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ChangeShopUserPasswordHandler.php:24]: It's unnecessary, as it has no effect. It's used by autoconfiguration which is disabled for bundles.

Comment (loic425) [reply src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ChangeShopUserPasswordHandler.php:24]: Yeah indeed, but Symfony uses it on Symfony\Component\Cache\Messenger\EarlyExpirationHandler.
Maybe it's just an information.

BTW, AsCommand attribute is now required, maybe the'll force using it at a moment.

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/composer.json [hunk 2]
```diff
@@ -54,6 +54,9 @@
     "extra": {
         "branch-alias": {
             "dev-main": "2.0-dev"
+        },
+        "symfony": {
+            "require": "7.0.*"
```
Comment (mpysiak) [inline src/Sylius/Bundle/ApiBundle/composer.json:59]:
```suggestion
            "require": "^7.0"
```

In every bundle

**Review by mpysiak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/composer.json (hunk 2 above)
Comment (GSadee) [reply src/Sylius/Bundle/ApiBundle/composer.json:59]: I would be even for bumping the minimum version to ^7.1, but definitely in a separate PR after merge this one 🖖🏻

**Review by Wojdylak — APPROVED** [review]

**Pull request merged**